### ISSUE TITLE
Add nullability annotations

### ISSUE THREAD

**Issue by HughIngram** [AltBeacon/android-beacon-library#788]
There are many methods within the api for this library which have nullable return types but no nullability annotation. The api would be a lot nicer to use (especially in Kotlin) if there were full coverage of the public methods.

The specific area where I have had issues is the method Beacon#getId2

**Comment by davidgyoung** [comment]
@HughIngram, agreed this would be good to add.   I would welcome a pull request if you would like to put one together.

**Comment by HughIngram** [comment]
@davidgyoung will do.
